Bottom: 362b9773c155c4eae92102eb184e462167809f06
Top:    8f1cb4cc02401b6df11cdcc7ed72d59196fe15d1
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-03-02 14:55:48 +0800

Refresh of vhost-prefetch-desc-indices

---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index 28f50ec..b04fa38 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -724,16 +724,13 @@ static void handle_rx(struct vhost_net *net)
 						&in, vq_log, &log, UIO_MAXIOV);
 		} else {
 			unsigned int out;
-			printk("cur %d ndescs %d\n", cur, ndescs);
 			if (cur == ndescs) {
 				npkts = sk_rx_array_length(sock->sk);
-				printk("npkts %d\n", npkts);
 				if (!npkts)
 					break;
 				ndescs = vhost_prefetch_desc_indices(vq,
 								indices,
 								min(npkts, 64));
-				printk("ndescs %d\n", ndescs);
 				cur = 0;
 				if (!ndescs) {
 					headcount = 0;
diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index cbe7f67..1bfb6fa 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -1915,7 +1915,6 @@ int vhost_prefetch_desc_indices(struct vhost_virtqueue *vq,
 	int ret;
 	u16 last_avail_idx, total, copied;
 	__virtio16 avail_idx;
-	__virtio16 *orig = indices;
 
 	if (unlikely(vhost_get_user(vq, avail_idx, &vq->avail->idx))) {
 		vq_err(vq, "Failed to access avail idx at %p\n",
@@ -1924,7 +1923,6 @@ int vhost_prefetch_desc_indices(struct vhost_virtqueue *vq,
 	}
 	vq->avail_idx = vhost16_to_cpu(vq, avail_idx);
 	total = vq->avail_idx - vq->last_avail_idx;
-	printk("num %d avail %u\n", num, total);
 	ret = total = min(num, total);
 
 	last_avail_idx = vq->last_avail_idx & (vq->num - 1);
@@ -1934,12 +1932,6 @@ int vhost_prefetch_desc_indices(struct vhost_virtqueue *vq,
 		copied = vq->num - last_avail_idx;
 		copied = min(total, copied);
 
-		if (indices + copied - orig > 64) {
-			printk("copied %d, before %d after %d\n",
-				copied, indices - orig,
-				indices + copied - orig);
-			return 0;
-		}
 		ret2 = vhost_copy_from_user(vq, indices,
 					    &vq->avail->ring[last_avail_idx],
 					    sizeof(avail_idx) * copied);
@@ -1956,7 +1948,6 @@ int vhost_prefetch_desc_indices(struct vhost_virtqueue *vq,
 	/* Only get avail ring entries after they have been exposed by guest. */
 	smp_rmb();
 
-	printk("ret is %d\n", ret);
 	return ret;
 }
 EXPORT_SYMBOL(vhost_prefetch_desc_indices);
